sound/soc/sof/intel/hda-stream.c

```diff
@@ -553,6 +553,23 @@ irqreturn_t hda_dsp_stream_interrupt(int irq, void *context)
 	return ret;
 }
 
+static void hda_dsp_set_bytes_transferred(struct hdac_stream *hstream)
+{
+	int pos, no_of_bytes;
+	unsigned int prev_pos;
+	u64 buffer_size = hstream->cstream->runtime->buffer_size;
+
+	div_u64_rem(hstream->curr_pos, buffer_size, &prev_pos);
+	pos = snd_hdac_stream_get_pos_posbuf(hstream);
```

Comment:
won't compile without SOF_HDA...

Reply:
Hmm, I did test on my machine this code with:
- all HDA devices enabled
- all APL-EHL enabled yet with "SOF support for HDA Links" disabled
- all HDA devices disabled

Compiles just fine. If I'm missing something (did not check some other option which hides the error away from me), please let me know.

Reply:
try running in nocodec mode.

Reply:
Took a look at SOF CI build-kernel results. Should be easier to navigate the problem now, will fix in next version.

Reply:
With v6, this should be fixed. Enlisted SND_SOC_SOF_HDA_PROBES to help me with that.